### ISSUE TITLE
Set_random_seed() does not work 

### ISSUE THREAD

**Issue by YubinXie** [clab/dynet#1365]
I initial my Dynet with the following codes, but I get different results every time from my LSTM model. Is this possible? 

`import _dynet as dy
dyparams = dy.DynetParams()
dyparams.set_mem(20000)
dyparams.set_random_seed(666)
dyparams.set_autobatch(True)
dyparams.init()`

**Comment by pmichel31415** [comment]
Hmm.. Are you getting this output?

```
[dynet] random seed: 666
[dynet] using autobatching
[dynet] allocating memory: 20000MB
[dynet] memory allocation done.
```

If yes, then the problem is outside of dynet (did you fix the random seed for other rngs ou might be using, like numpy.random ?)

If the random seed is different, can you try this syntax instead?

```python
import dynet_config
dynet_config.set(mem='20000', random_seed=666, autobatch=1)
import dynet as dy
```

**Comment by YubinXie** [comment]
I do get the above output. However, there is no numpy random involved in my codes. No other randomness. That is the reason that I feel weird . Will try you method, thanks.

**Comment by pmichel31415** [comment]
So FYI I've run this snippet several times:

```python
import _dynet as dy
dyparams = dy.DynetParams()
dyparams.set_mem(200)
dyparams.set_random_seed(666)
dyparams.set_autobatch(True)
dyparams.init()
dy.random_uniform(1, 0, 1).value()
```

And I've always gotten the same result (`0.7004371285438538`), could you test it on your setup?

Also, are you using a GPU?

**Comment by godweiyang** [comment]
> So FYI I've run this snippet several times:
> 
> ```python
> import _dynet as dy
> dyparams = dy.DynetParams()
> dyparams.set_mem(200)
> dyparams.set_random_seed(666)
> dyparams.set_autobatch(True)
> dyparams.init()
> dy.random_uniform(1, 0, 1).value()
> ```
> 
> And I've always gotten the same result (`0.7004371285438538`), could you test it on your setup?
> 
> Also, are you using a GPU?

I got the same issue as @YubinXie .
However, I have found that the initial values of the dynet tensors are the same. The first time that the tensors differ is after using `loss.backward()` and `trainer.update()`. So whether the problem results from the backpropogation?
